Source/Plugins/NetworkEvents/NetworkEvents.cpp

```diff
@@ -35,200 +35,75 @@ const int MAX_MESSAGE_LENGTH = 64000;
-}
-
-
-std::vector<String> StringTS::splitString (char sep)
-{
-    String S ((const char*)str, len);
-    String curr;
-
-    std::list<String> ls;
-    for (int k = 0; k < S.length(); ++k)
-    {
-        if (S[k] != sep)
-        {
-            curr+=S[k];
-        }
-        else
-        {
-            ls.push_back (curr);
-            while (S[k] == sep && k < S.length())
-                ++k;
-
-            curr = "";
-            if (S[k] != sep && k < S.length())
-                curr += S[k];
-        }
-    }
-    if (S.length() > 0)
-    {
-        if (S[S.length() - 1] != sep)
-            ls.push_back (curr);
-    }
-
-    std::vector<String> Svec (ls.begin(), ls.end());
-    return Svec;
-}
-
-
-StringTS::StringTS (MidiMessage& event)
-{
-    const uint8* dataptr = event.getRawData();
-    const int bufferSize = event.getRawDataSize();
-    len = bufferSize - 6 - 8; // -6 for initial event prefix, -8 for timestamp at the end
-
-    memcpy (&timestamp, dataptr + 6 + len, 8); // remember to skip first six bytes
-    str = new uint8[len];
-    memcpy (str,dataptr + 6, len);
-}
-
-
-StringTS& StringTS::operator= (const StringTS& rhs)
-{
-    delete (str);
-    len = rhs.len;
-    str = new uint8[len];
-    memcpy (str,rhs.str,len);
-    timestamp = rhs.timestamp;
-
-    return *this;
-}
-
-
-String StringTS::getString()
-{
-    return String ((const char*)str,len);
-}
-
-
-StringTS::StringTS (String S)
-{
-    Time t;
-    str = new uint8[S.length()];
-    memcpy (str, S.toRawUTF8(), S.length());
-    timestamp = t.getHighResolutionTicks();
-
-    len = S.length();
-}
-
-
-StringTS::StringTS (String S, int64 ts_software)
-{
-    str = new uint8[S.length()];
-    memcpy (str, S.toRawUTF8(), S.length());
-    timestamp = ts_software;
-
-    len = S.length();
-}
-
-
-StringTS::StringTS (const StringTS& s)
-{
-    str = new uint8[s.len];
-    memcpy (str, s.str, s.len);
-    timestamp = s.timestamp;
-    len = s.len;
-}
-
-
-StringTS::StringTS (unsigned char* buf, int _len, int64 ts_software) 
-    : len       (_len)
-    ,timestamp  (ts_software)
-{
-    str = new juce::uint8[len];
-    for (int k = 0; k < len; ++k)
-        str[k] = buf[k];
-}
-
-
-StringTS::~StringTS()
-{
-    delete str;
-}
-
-
-/*********************************************/
-void* NetworkEvents::zmqcontext = nullptr;
+NetworkEvents::ZMQContext* NetworkEvents::sharedContext = nullptr;
+CriticalSection NetworkEvents::sharedContextLock{};
 
 NetworkEvents::NetworkEvents()
     : GenericProcessor  ("Network Events")
     , Thread            ("NetworkThread")
-    , threshold         (200.0)
-    , bufferZone        (5.0f)
-    , state             (false)
+    , restart           (false)
+    , changeResponder   (false)
+    , urlport           (0)
 {
     setProcessorType (PROCESSOR_TYPE_SOURCE);
 
-    createZmqContext();
-
-    firstTime = true;
-    responder = nullptr;
-    urlport = 5556;
-    threadRunning = false;
+    if (!setNewListeningPort(5556))
```

Comment:
If we changed `setNewListeningPort` to not create the socket and return feedback right away, we couldn't do this, so maybe it would be better to just choose the port automatically by inputting 0. Of course loading a signal chain would still switch the port to the saved value, if possible.